### DISCUSSION TITLE
Use with Polylang Pro and DeepL-Integration

### DISCUSSION THREAD

**Discussion by dekaSW** [flyntwp/flynt#538] [Q&A]
Doing some tests with the most recent version of Polylang Pro that now integrated automatic translations via DeepL API I found that in order to make this feature work you need to change the field configuration of every field that should be translated.

So if you would like the content of a basic WYSIWYG-field to be automatically translated you'd need to add another option introduced by Polylang:

`            [
                'label' => __('Text', 'flynt'),
                'name' => 'contentHtml',
                'type' => 'wysiwyg',
                'delay' => 0,
                'media_upload' => 0,
                'required' => 1,
                'translations' => 'translate',  // add this to make automatic translation work
            ],`

see also:
https://polylang.pro/doc/working-with-acf-pro/#translate-fields-values
https://polylang.pro/deepl-machine-translation-now-available-for-polylang-pro/

I was wondering if this could be accomplished globally for specific field types via Flynts' ACF Field Group Composer.

**Comment by dekaSW** [comment]
Now on testing even further I found another issue that might keep me from using this setup overall.
In order for automatic translation to kick in, the field needs to be prepared as mentioned above
`'translations' => 'translate'`

But this has the side effect that as soon as only one field is setup this way, the translated post layouts are synchronized.
So a adding new flynt component block on a french post translation also gets synced to it's english counterpart.
Which makes things even worse is that this messes up the saved field data if this block is inserted somewhere in between or at the beginning.

This is due to the way polylang behaves on layout fields, see 4.1:
https://polylang.pro/doc/working-with-acf-pro/#translate-fields-values


@aaronmeder Have you found a fix/workaround for this already?

If this is not resolvable, Polylang Pro's new DeepL integration would not be compatible with flynt. :/

**Reply by aaronmeder** [reply]
@dekaSW not really, I haven't noticed that yet in my tests. For the moment we went on with WPML for a few reasons and did not continue more detailed testing with Polylang.
Maybe you could try to add a value of  "translations" => "copy_once" or similar to the flexible_content field itself in _inc/fieldGroups/pageComponents.php_?

**Reply by dekaSW** [reply]
Hey Aaaron,
thanks for your suggestion. Using "Copy_once" leaves the field unavailable for machine translation unfortunately.

If you continue your tests w/ Polylang, please keep me posted.

**Comment by dekaSW** [comment]
I might have come up with a fix for this issue (needs more testing though):

Exclude flynts' components from syncing via polylang filter hook:
(see https://polylang.pro/doc/filter-reference/)

```
// Polylang compat fix: do not sync flynt components in order to fix issues w/ machine translation
add_filter( 'pll_copy_post_metas', function ( $metas, $sync, $from, $to, $lang ) {

    if (! $sync) {
        return $metas;
    }

    // do not sync flyntwp layout components
    $excludes = [
        'pageComponents',
        'reusableComponents'
    ];
    // remove keys to be synced
    array_walk($excludes, function($key) use (&$metas) {
        if ( in_array($key, $metas)) {
            array_splice($metas, array_search($key, $metas), 1, null);
        }
    } );

    return $metas;
}, 10, 5);
```

This effectively leaves the component structure untouched when editing already translated content (for when field translation parameter is enabled for any of the contained fields). Yaih!
